client/src/utils/text.ts

```diff
@@ -39,7 +39,8 @@ export function cleanForSlug(
   // just do the "replace"
   if (unicodeSlugsEnabled) {
     return val
-      .replace(/\s/g, '-')
+      .trim()
```

Comment:
Adding trim here is not really going to work as expected

* It only will trim if `unicodeSlugsEnabled` and not when it is not enabled
* We probably should do the trim in the controller `slugify` method only, before passing it to this function, so that we do not break existing other behaviour that may exist